src/main/java/frc/trigon/robot/subsystems/ampaligner/AmpAligner.java

```diff
@@ -0,0 +1,118 @@
+import frc.trigon.robot.subsystems.pitcher.PitcherConstants;
+import org.trigon.hardware.RobotHardwareStats;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXMotor;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXSignal;
+
+public class AmpAligner extends MotorSubsystem {
+    private final TalonFXMotor motor = AmpAlignerConstants.MOTOR;
+    private final MotionMagicVoltage positionRequest = new MotionMagicVoltage(0).withEnableFOC(AmpAlignerConstants.FOC_ENABLED);
+    private final VoltageOut voltageRequest = new VoltageOut(0).withEnableFOC(AmpAlignerConstants.FOC_ENABLED);
+    private AmpAlignerConstants.AmpAlignerState targetState = AmpAlignerConstants.AmpAlignerState.CLOSE;
+    private Rotation2d targetAngle = Rotation2d.fromDegrees(0);
+
+    public AmpAligner() {
+        setName("AmpAligner");
+        configurePositionResettingTrigger();
+    }
+
+    @Override
+    public void stop() {
+        motor.stopMotor();
+    }
+
+    @Override
+    public void periodic() {
+        motor.update();
+        updateMechanism();
+    }
+
+    @Override
+    public void setBrake(boolean brake) {
+        motor.setBrake(brake);
+    }
+
+    @Override
+    public void drive(Measure<Voltage> voltageMeasure) {
+        motor.setControl(voltageRequest.withOutput(voltageMeasure.in(Units.Volts)));
+    }
+
+    @Override
+    public void updateLog(SysIdRoutineLog log) {
+        log.motor("AmpAligner")
+                .angularPosition(Units.Rotations.of(motor.getSignal(TalonFXSignal.POSITION)))
+                .angularVelocity(Units.RotationsPerSecond.of(motor.getSignal(TalonFXSignal.VELOCITY)))
+                .voltage(Units.Volts.of(motor.getSignal(TalonFXSignal.MOTOR_VOLTAGE)));
+    }
+
+    @Override
+    public SysIdRoutine.Config getSysIdConfig() {
+        return AmpAlignerConstants.SYSID_CONFIG;
+    }
+
+    public AmpAlignerConstants.AmpAlignerState getTargetState() {
+        return targetState;
+    }
+
+    public boolean atTargetState() {
+        return atAngle(targetState.targetAngle);
+    }
+
+    public boolean atTargetAngle() {
+        return atAngle(targetAngle);
+    }
+
+    public boolean atAngle(Rotation2d targetAngle) {
+        return Math.abs(getCurrentAngle().getDegrees() - targetAngle.getDegrees()) < AmpAlignerConstants.ANGLE_TOLERANCE.getDegrees();
+    }
+
+    public Rotation2d getCurrentAngle() {
+        return Rotation2d.fromRotations(motor.getSignal(TalonFXSignal.POSITION));
+    }
+
+    void setTargetState(AmpAlignerConstants.AmpAlignerState targetState) {
+        this.targetState = targetState;
+        setTargetAngle(targetState.targetAngle);
+    }
+
+    void setTargetAngle(Rotation2d targetAngle) {
+        this.targetAngle = targetAngle;
+        motor.setControl(positionRequest
+                .withPosition(targetAngle.getRotations())
+                .withFeedForward(calculateKGOutput())
+        );
+    }
+
+    private void configurePositionResettingTrigger() {
```

Comment:
I think this either needs a better name or a javadoc, I don't understand what it does